### PR TITLE
[Exporter.OpenTelemetryProtocol] fix event source throws `System.FormatException`

### PR DESCRIPTION
**Pull request by TimothyMothra**
This PR fixes a bug in the PersistentStorageEventSource class. 

```
System.FormatException
  HResult=0x80131537
  Message=Index (zero based) must be greater than or equal to zero and less than the size of the argument list.
  Source=System.Private.CoreLib
  StackTrace:
   at System.ThrowHelper.ThrowFormatIndexOutOfRange()
   at System.Text.ValueStringBuilder.AppendFormatHelper(IFormatProvider provider, String format, ReadOnlySpan`1 args)
   at System.String.FormatHelper(IFormatProvider provider, String format, ReadOnlySpan`1 args)
   at System.String.Format(IFormatProvider provider, String format, Object[] args)
   at Microsoft.Azure.WebJobs.Script.WebHost.Diagnostics.OpenTelemetryEventListener.OnEventWritten(EventWrittenEventArgs eventData) in C:\Users\Rohit Ranjan\source\repos\azure-functions-host\src\WebJobs.Script.WebHost\Diagnostics\OpenTelemetryEventListener.cs:line 54
   at System.Diagnostics.Tracing.EventSource.DispatchToAllListeners(EventWrittenEventArgs eventCallbackArgs)
```

### Changes
- fix index in Event attribute.
- Update test class to verify all event source classes in this project.

### CODE REVIEW

File: test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/EventSourceTest.cs [hunk 1]
```diff
@@ -14,4 +16,16 @@ public void EventSourceTest_OpenTelemetryProtocolExporterEventSource()
     {
         EventSourceTestHelper.MethodsAreImplementedConsistentlyWithTheirAttributes(OpenTelemetryProtocolExporterEventSource.Log);
     }
+
+    [Fact]
+    public void EventSourceTest_PersistentStorageAbstractionsEventSource()
```
Comment (cijothomas) [inline test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/EventSourceTest.cs:21]: Thanks for fixing this!

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+* Fix `System.FormatException`.
```
Comment (cijothomas) [inline src/OpenTelemetry.Exporter.OpenTelemetryProtocol/CHANGELOG.md:5]: please use a more descriptive changelog.

**Review by cijothomas — APPROVED** [review]

**Review by reyang — APPROVED** [review]
Another good reason why to move to ILogger #3881.

**Review by rajkumar-rangaraj — APPROVED** [review]

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+* Fix `System.FormatException` thrown by `PersistentStorageEventSource`.
```
Comment (vishweshbankwar) [inline src/OpenTelemetry.Exporter.OpenTelemetryProtocol/CHANGELOG.md:5]: No changelog is needed here. This component is not yet used by any code path. The changelog needs to be added [here](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/tree/main/src/OpenTelemetry.PersistentStorage.FileSystem)

**Review by vishweshbankwar — APPROVED** [review]

**Review by utpilla — APPROVED** [review]

**Pull request merged**